### ISSUE TITLE
support for powerpc architecture

### ISSUE THREAD

**Issue by Pintu79** [akopytov/sysbench#129]
I tried to build it on my machine which is powerpc 64 bit little endian. I have ubuntu 16.04 OS.
I got the following error-  #error "No support for PowerPC 64 bit mode". I also tried with replacing the luajit lib folder with downloading the current one from github, but i am not succesful to build it. Is some extra steps required to build it on power pc architecture or is it missing the feature for powerpc

**Comment by akopytov** [comment]
Yes, unfortunately upstream LuaJIT has no support for PPC64 at the moment. Can you try replacing bundled LuaJIT with https://github.com/PPC64/LuaJIT? If it is functional, I will at least make sure to document it as an option. Thanks.

**Comment by Pintu79** [comment]
replacing with this LuaJIT link, sysbench is working for me. below is the snapshot.
ci@t2ctl:~/sysbench.install/bin$ ./sysbench --test=cpu --num-threads=80 --cpu-max-prime=500000 run
WARNING: the --test option is deprecated. You can pass a script name or path on the command line without any options.
WARNING: --num-threads is deprecated, use --threads instead
sysbench 1.1.0-ec17d39 (using bundled LuaJIT 2.1.0-beta2)

Running the test with following options:
Number of threads: 80
Initializing random number generator from current time


Prime numbers limit: 500000

Initializing worker threads...

Threads started!


General statistics:
    total time:                          10.2605s
    total number of events:              1877

Latency (ms):
         min:                                 97.36
         avg:                                429.66
         max:                               1074.42
         95th percentile:                    694.45
         sum:                             806473.70

Threads fairness:
    events (avg/stddev):           23.4625/3.77
    execution time (avg/stddev):   10.0809/0.10
many thanks akopytov

**Comment by akopytov** [comment]
@Pintu79 may I ask you to run 'make test' on your machine as well to make sure Lua API is functional?

**Comment by Pintu79** [comment]
I have run the test, but one of the test case is failed. So, is it necessary to pass all the test cases. I run the sysbench for mysql workload also. It was running smoothly. I run it with oltp-read-only.lua and oltp-write-only.lua. It was running good. So, should I believe the result what i get after run. below is the snapshot of make test -
ci@t2ctl:~/sysbench$ make test
cd tests && make test
make[1]: Entering directory '/home/ci/sysbench/tests'
./test_run.sh
t/1st.t: passed
t/api_basic.t: passed
t/api_histogram.t: passed
t/api_legacy_basic.t: skipped
t/api_legacy_rand.t: passed
t/api_legacy_sql.t: skipped
t/api_rand.t: passed
t/api_reports.t: passed
t/api_sql_mysql.t: skipped
t/api_sql_pgsql.t: skipped
t/cmd_cleanup.t: passed
t/cmd_help.t: passed
t/cmdline.t: passed
t/cmd_prepare.t: passed
t/cmd_run.t: passed
t/commands.t: passed
t/drivers.t: passed
t/drv_mysql.t: skipped
t/drv_pgsql.t: skipped
t/help_drv_mysql.t: passed
t/help_drv_pgsql.t: skipped
t/opt_help.t: passed
t/opt_histogram.t: passed
t/opt_luajit_cmd.t: failed
--- t/opt_luajit_cmd.t
+++ t/opt_luajit_cmd.t.err
@@ -8,7 +8,7 @@
   $ sysbench --luajit-cmd opt_luajit_cmd.lua
   sysbench * (glob)

-  JIT status: on
+  JIT status: off

   $ sysbench --luajit-cmd=off opt_luajit_cmd.lua
   sysbench * (glob)
@@ -18,4 +18,5 @@
   $ sysbench --luajit-cmd=on opt_luajit_cmd.lua
   sysbench * (glob)

-  JIT status: on
+  FATAL: `lua_pcall' function failed: no JIT compiler for this architecture (yet)
+  [1]
t/opt_report_checkpoints.t: passed
t/opt_report_interval.t: passed
t/opt_version.t: passed
t/opt_warmup_time.t: passed
t/script_bulk_insert_mysql.t: skipped
t/script_bulk_insert_pgsql.t: skipped
t/script_oltp_help.t: passed
t/script_oltp_insert_mysql.t: skipped
t/script_oltp_insert_pgsql.t: skipped
t/script_oltp_point_select_mysql.t: skipped
t/script_oltp_point_select_pgsql.t: skipped
t/script_oltp_read_write_mysql.t: skipped
t/script_oltp_read_write_pgsql.t: skipped
t/script_select_random_mysql.t: skipped
t/script_select_random_pgsql.t: skipped
t/test_cpu.t: passed
t/test_fileio.t: passed
t/test_memory.t: passed
t/test_mutex.t: passed
t/tests.t: passed
t/test_threads.t: passed
# Ran 45 tests, 17 skipped, 1 failed.
Makefile:963: recipe for target 'test' failed
make[1]: *** [test] Error 1
make[1]: Leaving directory '/home/ci/sysbench/tests'
Makefile:913: recipe for target 'test' failed
make: *** [test] Error 2
Kind Regards,
  Pintu

**Comment by akopytov** [comment]
Thanks, that failure is expected. I'll see what can be done to support PPC64 out of the box.

**Comment by akopytov** [comment]
I think I can fix PPC64 at least partially, but I need access to a ppc64le machine. I'm also happy to accept patches.
